### PR TITLE
[feature] s3 account public access block

### PR DESCRIPTION
**Pull request by edulop91**
### Summary
A module to help configure account wide S3 public access blocks.

### Test Plan
Unittests. Note that a quirk about testing here is that this resource is global to an account. If there are multiple resources configuring account wide blocks, they will conflict for each other. A small hack I made to work around this is to output the values (true/false) that we pass to the configuration block so we can assert on those instead.

### References
https://docs.aws.amazon.com/AmazonS3/latest/API/API_control_PublicAccessBlockConfiguration.html

### CODE REVIEW

File: aws-s3-account-public-access-block/main.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  # These affect existing buckets, policies, and acls
+  ignore_public_acls      = local.is_all
+  restrict_public_buckets = local.is_all
+
+}
+
+
+# See https://docs.aws.amazon.com/AmazonS3/latest/API/API_control_PublicAccessBlockConfiguration.html for more info
+resource "aws_s3_account_public_access_block" "block" {
```
Comment (cziie) [inline aws-s3-account-public-access-block/main.tf:20]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing version constraint for provider "aws" in "required_providers"

File: aws-s3-account-public-access-block/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+locals {
+  is_all  = var.restrict == "all"
+  is_new  = var.restrict == "new"
```
Comment (cziie) [inline aws-s3-account-public-access-block/main.tf:3]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>local.is_new is declared but not used

File: aws-s3-account-public-access-block/Makefile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Auto-generated by fogg. Do not edit
```
Comment (ryanking) [inline aws-s3-account-public-access-block/Makefile:1]: I think you can rm this file.

File: aws-s3-account-public-access-block/fogg.tf [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Auto-generated by fogg. Do not edit
```
Comment (ryanking) [inline aws-s3-account-public-access-block/fogg.tf:1]: rm

**Review by ryanking — CHANGES_REQUESTED** [review]

**Review by ryanking — APPROVED** [review]

**Pull request merged**